/**
 * @file admin_page.h
 * @brief Заголовочный файл административной панели
 * @author Команда разработчиков
 * @date 2025
 * @version 1.0
 * 
 * Содержит объявление класса admin_page, который предоставляет
 * интерфейс для администрирования системы казино.
 */

#ifndef ADMIN_PAGE_H
#define ADMIN_PAGE_H

#include <QDialog>
#include <QTableWidget>
#include <QLineEdit>
#include <QPushButton>

namespace Ui {
class admin_page;
}

/**
 * @brief Административная панель для управления системой казино
 * 
 * Класс admin_page представляет интерфейс администратора, который позволяет:
 * - Просматривать статистику всех пользователей
 * - Управлять балансами пользователей
 * - Пополнять счета игроков
 * - Мониторить активность в системе
 * 
 * Доступ к этой панели имеют только пользователи с правами администратора.
 * Интерфейс построен на основе диалогового окна с таблицей пользователей
 * и элементами управления для административных операций.
 */
class admin_page : public QDialog
{
    Q_OBJECT

public:
    /**
     * @brief Конструктор административной панели
     * @param parent Родительский виджет (по умолчанию nullptr)
     * 
     * Инициализирует интерфейс админ-панели и настраивает
     * таблицу для отображения данных пользователей.
     */
    explicit admin_page(QWidget *parent = nullptr);
    
    /**
     * @brief Деструктор административной панели
     * 
     * Освобождает ресурсы, используемые админ-панелью.
     */
    ~admin_page();
    
    /**
     * @brief Отображение статистики пользователей
     * 
     * Загружает и отображает в таблице информацию о всех
     * зарегистрированных пользователях, включая их балансы
     * и игровую статистику.
     */
    void viewStatistics();
    
    /**
     * @brief Слот для показа административной панели
     * 
     * Отображает админ-панель и обновляет данные статистики.
     * Вызывается при переходе к админ-разделу из главного меню.
     */
    void slot_show();

private slots:
    /**
     * @brief Слот для обработки нажатия кнопки "Пополнить баланс"
     * 
     * Выполняет операцию пополнения баланса указанного пользователя
     * на заданную сумму. Проверяет корректность введенных данных
     * и обновляет информацию в базе данных.
     */
    void addMoneyToUser();

private:
    /** @brief Указатель на UI объект административной панели */
    Ui::admin_page *ui;

    /** @brief Таблица для отображения списка пользователей и их статистики */
    QTableWidget *tableWidget;
    
    /** @brief Поле ввода имени пользователя для операций с балансом */
    QLineEdit *user_to_send;
    
    /** @brief Поле ввода суммы для пополнения баланса пользователя */
    QLineEdit *amount_to_send;
};

#endif // ADMIN_PAGE_H
